xarray/backends/zarr.py

```diff
@@ -416,6 +415,104 @@ class ZarrStore(AbstractWritableDataStore):
         "_close_store_on_close",
     )
 
+    @classmethod
+    def open_store(
```

Comment:
Could you rewrite `open_group` to call `open_store()` internally? That would reduce the amount of duplicated code and make this easier to maintain going forward.